packages/react-storage/src/components/StorageBrowser/adapters/createAmplifyAuthAdapter.ts

```diff
@@ -21,7 +22,7 @@ const isTemporaryCredentials = (
   !!value?.sessionToken || !!value?.expiration;
 
 export const createAmplifyAuthAdapter = (input?: {
-  options?: { defaultPrefixes?: string[] };
+  options?: { defaultPrefixes?: (string | ((identityId: string) => string))[] };
```

Comment:
What do you think of providing the bucket/buckets as the param to the callback along with the `identityId`? That would allow conditional logic per bucket

Reply:
Made a follow up task to handle with multibucket support